cli/src/commands/chain_spec/fetch.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use jsonrpsee::{
+    async_client::ClientBuilder,
+    client_transport::ws::WsTransportClientBuilder,
+    core::{client::ClientT, Error},
+    http_client::HttpClientBuilder,
+};
+use std::time::Duration;
+use subxt_codegen::fetch_metadata::Url;
+
+/// Returns the node's chainSpec from the provided URL.
+pub async fn fetch_chain_spec(url: Url) -> Result<serde_json::Value, FetchSpecError> {
+    async fn fetch_ws(url: Url) -> Result<serde_json::Value, Error> {
```

Comment:
This is clean but my hunch is that we have duplicated this in a few places now, maybe a util/helper crate for this is worth it? 

No need to address it in this PR but just a thought, worth filing an issue IMO :)